applications/tests/function_tests/test_cuda_geo_transformer_produce_and_load_coeffs.cpp

```diff
@@ -0,0 +1,297 @@
+#include <gtest/gtest.h>
+#include <memory>
+#include <random>
+
+#include "reconstruction_cuda/cuda_gpu_geo_transformer.h"
+
+template< typename T >
+class GeoTransformerTest : public GeoTransformer< T > {
+public:
+
+    template< typename IN >
+    void produceAndLoadCoeffsTest( const MultidimArray<IN>& input ) {
+        GeoTransformer<T>::produceAndLoadCoeffs( input );
+    }
+
+    std::unique_ptr<T[]> copy_out_d_inTest( size_t size ) const {
+        return GeoTransformer<T>::copy_out_d_in( size );
+    }
+};
+
+template< typename T >
+class GeoTransformerProduceAndLoadCoeffsTest : public ::testing::Test {
+
+public:
+
+    void compare_results( T* true_values, T* approx_values, size_t size ) {
+        for ( size_t i = 0; i < size; ++i ) {
+            ASSERT_NEAR( true_values[i], approx_values[i], static_cast< T >( 0.00001 ) ) << "at index=" << i;
+        }
+    }
+
+    void allocate_arrays() {
+        size = x * y;
+        in = std::unique_ptr<T[]>( new T[size]() );
+        out = std::unique_ptr<T[]>( new T[size]() );
+    }
+
+    void run_test() {
+        cpu_reference( in.get(), out.get(), x, y );
+
+        MultidimArray< T > tmpIn( y, x );
+        FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY( tmpIn ) {
```

Comment:
if you don't want to copy the array, you can simply change the pointer (tmpIn.data = in)

Reply:
Fixed.